hazelcast/src/main/java/com/hazelcast/client/impl/spi/ProxyManager.java

```diff
@@ -330,13 +388,58 @@ private void initialize(ClientProxy clientProxy) throws Exception {
     }
 
     public Collection<? extends DistributedObject> getDistributedObjects() {
+        getAndProcessDistributedObjects();
```

Comment:
I still don't think that a getter method should actually do more than getting but modifying the local state. Why don't we just get the remote and combine with what is in local without actually creating or destroying proxies. Hence, it is a read-only method.

Furthermore, if there are 1000 objects in cluster but this client uses only one of them, why would I create unused 999 objects in my local and try to keep them synchronized?